compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -477,6 +489,20 @@ class ComposeScene internal constructor(
         forEachOwner { it.measureAndLayout() }
         pointerPositionUpdater.beforeEvent(event)
         processPointerInput(event)
+        updatePointerPositions(event)
+    }
+
+    private fun updatePointerPositions(event: PointerInputEvent) {
+        // touches/styluses positions should be removed from [pointerPositions] if they are not down anymore
+        val releasedPointers = event.pointers.filter { it.type != PointerType.Mouse && !it.down }
+        for (pointer in releasedPointers) {
+            pointerPositions.remove(pointer.id)
+        }
+        // update positions for pointers that are down + mouse
+        val pointersToUpdate = event.pointers - releasedPointers.toSet()
+        for (pointer in pointersToUpdate) {
+            pointerPositions[pointer.id] = pointer.position
+        }
```

Comment:
Hm, good point about frequency. Now I am afraid that it is needed only for popups (and not sure that it will be needed somewhere else) and we provide such overhead on any pointer events everywhere.

> Also, do we have a sparse-array implementation available?

I don't know any default implementations for it (only in android)

Reply:
I've profiled it and with 1ms sampling I see that PR's implementation adds max 1% overhead on `sendPointerEvent` call, so I'll convert it to your implementation with less temporary objects, profile and if it is better to the same, I'll push it